core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala

```diff
@@ -190,19 +189,8 @@ private[deploy] class Worker(
 
   private def createWorkDir() {
     workDir = Option(workDirPath).map(new File(_)).getOrElse(new File(sparkHome, "work"))
-    try {
-      // This sporadically fails - not sure why ... !workDir.exists() && !workDir.mkdirs()
-      // So attempting to create and then check if directory was created or not.
-      workDir.mkdirs()
-      if ( !workDir.exists() || !workDir.isDirectory) {
-        logError("Failed to create work directory " + workDir)
-        System.exit(1)
-      }
-      assert (workDir.isDirectory)
-    } catch {
-      case e: Exception =>
-        logError("Failed to create work directory " + workDir, e)
-        System.exit(1)
+    if (!Utils.createDirectory(workDir)) {
```

Comment:
Ur, `Utils.createDirectory` works differently from `workDir.mkdirs()`.
Do we need to change the current behavior of `createWorkDir` fo this PR?
This behavior change may break 3rd party tools.